core/base/planarGraphLayout/MergeTreeVisu.h

```diff
@@ -0,0 +1,2060 @@
+    allBaryPercentMatch = baryPercentMatch;
+  }
+
+  // Temporal Subsampling Output
+  void setInterpolatedTrees(std::vector<bool> &isInterpolatedTrees) {
+    interpolatedTrees = isInterpolatedTrees;
+  }
+
+  // Output
+  void setVtkOutputNode(vtkSmartPointer<vtkUnstructuredGrid> vtkNode) {
+    vtkOutputNode = vtkNode;
+  }
+  void setVtkOutputArc(vtkSmartPointer<vtkUnstructuredGrid> vtkArc) {
+    vtkOutputArc = vtkArc;
+  }
+  void setVtkOutputSegmentation(
+    vtkSmartPointer<vtkUnstructuredGrid> vtkSegmentation) {
+    vtkOutputSegmentation = vtkSegmentation;
+  }
+
+  // Matching output
+  void setVtkOutputNode1(vtkSmartPointer<vtkUnstructuredGrid> vtkNode1) {
+    vtkOutputNode1 = vtkNode1;
+  }
+  void setVtkOutputNode2(vtkSmartPointer<vtkUnstructuredGrid> vtkNode2) {
+    vtkOutputNode2 = vtkNode2;
+  }
+  void setNodeCorr1(std::vector<std::vector<SimplexId>> &nodeCorrT) {
+    nodeCorr1 = nodeCorrT;
+  }
+  void setNodeCorr2(std::vector<std::vector<SimplexId>> &nodeCorrT) {
+    nodeCorr2 = nodeCorrT;
+  }
+  void setVtkOutputMatching(vtkSmartPointer<vtkUnstructuredGrid> vtkMatching) {
+    vtkOutputMatching = vtkMatching;
+  }
+  void setOutputMatching(
+    std::vector<std::tuple<ftm::idNode, ftm::idNode, double>> &matching) {
+    outputMatchingBarycenter = std::vector<
+      std::vector<std::vector<std::tuple<ftm::idNode, ftm::idNode>>>>(1);
+    outputMatchingBarycenter[0]
+      = std::vector<std::vector<std::tuple<ftm::idNode, ftm::idNode>>>(1);
+
+    std::vector<std::tuple<ftm::idNode, ftm::idNode>> tMatching;
+    for(auto match : matching)
+      tMatching.push_back(
+        std::make_tuple(std::get<0>(match), std::get<1>(match)));
+
+    outputMatchingBarycenter[0][0] = tMatching;
+  }
+
+  // Filled by the algorithm
+  std::vector<std::vector<SimplexId>> getNodeCorr() {
+    return nodeCorr;
+  }
+
+  std::vector<double> getClusterShift() {
+    return clusterShift;
+  }
+
+  // ==========================================================================
+  // Matching Visualization
+  // ==========================================================================
+  template <class dataType>
+  void
+    makeMatchingOutput(FTMTree_MT *tree1, FTMTree_MT *tree2, int verbose = 0) {
+    std::vector<FTMTree_MT *> trees{tree1, tree2};
+    std::vector<FTMTree_MT *> barycenters;
+
+    makeMatchingOutput<dataType>(trees, barycenters, verbose);
+  }
+
+  template <class dataType>
+  void makeMatchingOutput(std::vector<FTMTree_MT *> &trees,
+                          std::vector<FTMTree_MT *> &barycenters,
+                          int verbose = 0) {
+    // verbose = 2;
+
+    int numInputs = trees.size();
+    int NumberOfBarycenters = barycenters.size();
+    bool clusteringOutput = (NumberOfBarycenters != 0);
+    NumberOfBarycenters
+      = std::max(NumberOfBarycenters, 1); // to always enter the outer loop
+    if(not clusteringOutput)
+      numInputs = 1;
+
+    vtkSmartPointer<vtkUnstructuredGrid> vtkMatching
```

Comment:
yikes :(
this is unfortunately breaking the API structure :(
the base code classes should *NOT* depend on VTK!

Reply:
@pierre-guillou @CharlesGueunet 
the fact that the CI let that go through is a little concerning.
we may want to enforce that test (for instance, by building the vtk-less version of the library in the CI).

Reply:
Well, it's difficult if the faulty headers are only included by source files in the VTK layer and not in the base layer, as this seems to be the case here. I'm afraid that even if we build TTK without VTK (only the base layer), these headers won't be included anywhere and there won't be any build error (I need to test that though).

One way to detect that is to automatically generate a header that includes every base layer header and use it, for instance, in the c++ base-only example. Unfortunately, due to the modularity of the base layer, this might be hard to automate.

Reply:
Yes, actually at first it was in the vtk layer, but then I moved it in the base layer when merging with planarGraphLayout, we could split the file or let it in the vtk layer. Like you said, we wil discuss about it soon!